### PR TITLE
Add support for ximalaya

### PR DESCRIPTION
**Pull request by KevinSource0**
Add support for ximalaya, download audio by url.
Potential future improvement: download album.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -621,6 +621,7 @@ $ lux -j "https://www.bilibili.com/video/av20203945"
 | Eporner    | <https://eporner.com>        | ✓        |          |            |                  |
 | StreamTape | <https://streamtape.com>     | ✓        |          |            |                  |
 | 虎扑       | https://hupu.com             | ✓        |          |            |                  |
+| 喜马拉雅   | https://www.ximalaya.com/    | ✓        |          |            |                  |
```
Comment (iawia002) [inline README.md:624]: You can add a new column of the audio type here(after the Images column)

File: extractors/ximalaya/ximalaya.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	var ximalaya ximalayaData
+	if err = json.Unmarshal([]byte(jsonData), &ximalaya); err != nil {
+		return nil, err
+	}
+
+	realURL := ximalaya.Data.Src
+	urlData := make([]*types.Part, 0)
+	var totalSize int64
+
+	totalSize, err = request.Size(realURL, url)
```
Comment (iawia002) [inline extractors/ximalaya/ximalaya.go:56]:
```suggestion
	totalSize, err := request.Size(realURL, url)
```

**Review by iawia002 — COMMENTED** [review]
Looks pretty good, just a few comments.

File: README.md (hunk 1 above)
Comment (KevinSource0) [reply README.md]: sure, added a audio column

File: extractors/ximalaya/ximalaya.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	urlData := make([]*types.Part, 0)
+	totalSize, err := request.Size(realURL, url)
+	if err != nil {
+		return nil, err
+	}
+
+	urlData = append(urlData, &types.Part{
+		URL:  realURL,
+		Size: totalSize,
+		Ext:  "m4a",
```
Comment (iawia002) [inline extractors/ximalaya/ximalaya.go:62]: One more question, is this file extension fixed and can we use the file extension on the download link?

Comment (iawia002) [reply extractors/ximalaya/ximalaya.go]: Never mind, I have updated this in d04d2ff

**Pull request merged**